pkg/utils/kubernetes/featuregates.go

```diff
@@ -0,0 +1,214 @@
+// Copyright (c) 2021 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetes
+
+import (
+	"fmt"
+
+	"github.com/gardener/gardener/pkg/utils/version"
+)
+
+// featureGateVersionRanges contains the version ranges for all Kubernetes feature gates.
+// Extracted from https://raw.githubusercontent.com/kubernetes/kubernetes/release-${version}/pkg/features/kube_features.go.
+// To maintain this list for each new Kubernetes version:
+// * Run hack/compare-k8s-feature-gates.sh <old-version> <new-version>.
+//   It will present 2 lists of feature gates: those added and those removed in <new-version> compared to <old-version>.
+// * Add all added feature gates to the map with <new-version> as MinVersion and no MaxVersion.
+// * For any removed feature gates, add <new-version> as MaxVersion to the already existing feature gate in the map.
+var featureGateVersionRanges = map[string]*version.Range{
```

Comment:
Can you add the special `AllAlpha` and `AllBeta` feature gates ([ref](https://github.com/kubernetes/kubernetes/blob/5c137f1af94e23a71e7a7a3dad037f1193147778/staging/src/k8s.io/component-base/featuregate/feature_gate.go#L38-L48))?
Otherwise, endusers won't be able to enable them.
I think, they were added in `1.17` (ref https://github.com/kubernetes/kubernetes/pull/84304).

Reply:
Yes, I will add them, thanks for noticing.

Reply:
Done